src/NUnitTestAdapter/NUnitTestAdapter.cs

```diff
@@ -81,16 +56,51 @@ public static bool IsRunningUnderIDE
 
         #region Protected Helper Methods
 
+        // The Adapter is constructed using the default constructor.
+        // We don't have any info to initialize it until one of the
+        // ITestDiscovery or ITestExecutor methods is called. The
+        // each Discover or Execute method must call this method.
+        protected virtual void Initialize(IMessageLogger messageLogger)
+        {
+            _verbosity = 0; // In case we throw below
+            _adapterVersion = Assembly.GetExecutingAssembly().GetName().Version.ToString();
+
+            try
+            {
+                var registry = RegistryCurrentUser.OpenRegistryCurrentUser(@"Software\nunit.org\VSAdapter");
+                UseVsKeepEngineRunning = registry.Exist("UseVsKeepEngineRunning") && (registry.Read<int>("UseVsKeepEngineRunning") == 1);
+                _shadowCopy = registry.Exist("ShadowCopy") && (registry.Read<int>("ShadowCopy") == 1);
+                _verbosity = (registry.Exist("Verbosity")) ? registry.Read<int>("Verbosity") : 0;
+            }
+            catch (Exception e)
+            {
+                messageLogger.SendMessage(TestMessageLevel.Error, e.ToString());
+                // TODO: Shouldn't we rethrow here?
```

Comment:
I'll remove comment and add a message that says "Default settings will be used."